boot/bootutil/src/bootutil_misc.c

```diff
@@ -615,10 +615,16 @@ boot_set_pending(int permanent)
         flash_area_close(fap);
         return rc;
 
+    case BOOT_MAGIC_BAD:
+        /* The image slot is corrupt.  There is no way to recover, so erase the
+         * slot to allow future upgrades.
+         */
+        flash_area_erase(fap, 0, fap->fa_size);
```

Comment:
It is weird that you are not calling `flash_area_open` so no idea where `fap` is initialized.

Reply:
Indeed. This is actually a double fail on my part - I meant to test this in the morning before submitting the PR, then forgot to test it!

Thanks, @utzig.  I will fix and test :)